scene/2d/tile_map.cpp

```diff
@@ -194,13 +194,16 @@ void TileMapLayer::_rendering_update() {
 		}
 	} else {
 		// Create/Update the layer's CanvasItem.
-		if (!canvas_item.is_valid()) {
-			RID ci = rs->canvas_item_create();
+		RID &ci = canvas_item;
+		if (ci.is_null()) {
+			ci = rs->canvas_item_create();
 			rs->canvas_item_set_parent(ci, tile_map_node->get_canvas_item());
+			dirty.flags[DIRTY_FLAGS_LAYER_INDEX_IN_TILE_MAP_NODE] = true;
+		}
+		if (dirty.flags[DIRTY_FLAGS_LAYER_INDEX_IN_TILE_MAP_NODE]) {
 			rs->canvas_item_set_draw_index(ci, layer_index_in_tile_map_node - (int64_t)0x80000000);
```

Comment:
I do'nt think you need that many change. Just move `rs->canvas_item_set_draw_index(ci, layer_index_in_tile_map_node - (int64_t)0x80000000);` out of the `if (!canvas_item.is_valid()) {` condition. I think it's fine to update this at every update since nothing else is optimized here anyway.

Reply:
Ok made it like suggested, just a note that `DIRTY_FLAGS_LAYER_INDEX_IN_TILE_MAP_NODE` is currently pointeless / not used.

Reply:
> not really, there are other flags like that. They are used by the debug view anyway, as any of the flags set to true will trigger a redraw.

Oh, right, I've checked only for the direct usage of this specific flag. :man_facepalming:

> Basically, I could have created a generic "SOMETHING CHANGED" flag instead, but I think it's simply easier to be able to know what was modified instead of tracking back where the change was done.

Yeah, makes sense.